### PR TITLE
multi stage docker build, semver tagging

### PR DESCRIPTION
**Pull request by mgmgwi**
I would like to start by apologizing for the fact that the PR contains several unrelated changes:

1. The image tagging is altered to a [semver](https://semver.org/)-like versioning by using the `docker/metadata-action@v3` action. This means to get a `latest` tag, we need to push a tag like `v0.1.0` to GitHub. [semver](https://semver.org/) recommends to start the initial development release at `v0.1.0`.
2. The indy_node image is now build, with a multi-stage build. This should optimize the image size and limit possible attack vectors further. Please check this carefully and give feedback. See build/Dockerfile.buster and build/Dockerfile.ubuntu18
3. The build actions have now limited caching for Docker image layers.
4. There is a `Makefile` for convenient local building.
5. The `Makefile` contains a `check_buster` and a `check_ubuntu` target, which runs a local `trivy image` check. This requires a local [trivy](https://aquasecurity.github.io/trivy/) installation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	docker build -f "build/Dockerfile.buster" -t indy-node-container/indy_node:buster --build-arg BASE=python:3.6-slim-buster --build-arg BUILDER_BASE=indy-node-container/base:buster build
+
+ubuntu_base:
+	docker build -f "build/Dockerfile.base.ubuntu18" -t indy-node-container/base:ubuntu build
+
+ubuntu: ubuntu_base
+	docker build -f "build/Dockerfile.ubuntu18" -t indy-node-container/indy_node:ubuntu --build-arg BASE=ubuntu:18.04 --build-arg BUILDER_BASE=indy-node-container/base:ubuntu build
+
+clean_buster_base:
+	-docker image rm indy-node-container/base:buster
```
Comment (solidnerd) [inline Makefile:16]: Why do we use here `-<COMMAND>` is this something related to Makefiles?  Or only a typo ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+clean_buster:
+	-docker image rm indy-node-container/indy_node:buster
```
Comment (solidnerd) [inline Makefile:19]: See Comment above

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+clean_ubuntu_base:
+	-docker image rm indy-node-container/base:ubuntu
```
Comment (solidnerd) [inline Makefile:22]: See Comment above

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+clean_ubuntu:
+	-docker image rm indy-node-container/indy_node:ubuntu
```
Comment (solidnerd) [inline Makefile:25]: See Comment above

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+clean: clean_buster clean_buster_base clean_ubuntu clean_ubuntu_base
+
+
+# all check targets require a local trivy installation - see https://aquasecurity.github.io/trivy/
+
+check_buster:
+	mkdir -p trivy-reports
+	-trivy image --format template --template "@trivy/html.tpl" -o trivy-reports/buster.html indy-node-container/indy_node:buster
```
Comment (solidnerd) [inline Makefile:34]: See Comment above

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#	-xdg-open trivy-reports/buster.html
+	-trivy image --severity HIGH,CRITICAL indy-node-container/indy_node:buster
```
Comment (solidnerd) [inline Makefile:36]: See Comment above

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+check_ubuntu:
+	mkdir -p trivy-reports
+	-trivy image --format template --template "@trivy/html.tpl" -o trivy-reports/ubuntu.html indy-node-container/indy_node:ubuntu
```
Comment (solidnerd) [inline Makefile:40]: See Comment above

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+#	-xdg-open trivy-reports/ubuntu.html
+	-trivy image --severity HIGH,CRITICAL indy-node-container/indy_node:ubuntu
```
Comment (solidnerd) [inline Makefile:42]: See Comment above

**Comment by solidnerd** [comment]
@mgmgwi  Should add also right now a Contributing.md and there mentions for the local prerequisites like trivy and co. But that a separate Task.

**Comment by solidnerd** [comment]
All the Rest looks fine to me.

File: Makefile (hunk 1 above)
Comment (mgmgwi) [reply Makefile:16]: the `-` means: "continue on error/non-zero exit", which occurs if the image is missing

**Comment by mgmgwi** [comment]
> @mgmgwi Should add also right now a Contributing.md and there mentions for the local prerequisites like trivy and co. But that a separate Task.

I created issue https://github.com/IDunion/indy-node-container/issues/19 for that.

File: Makefile (hunk 1 above)
Comment (solidnerd) [reply Makefile:16]: Ah that's cool. I used mostly `COMMAND || /bin/true`

**Review by solidnerd — APPROVED** [review]
